src/humanloop/overload.py

```diff
@@ -114,11 +118,105 @@ def _overload_call(self, **kwargs) -> PromptCallResponse:
             raise HumanloopRuntimeError from e
 
         return response
 
     # Replace the original log method with the overloaded one
     client.call = types.MethodType(_overload_call, client)  # type: ignore [assignment]
     return client
+
+def _get_file_type_from_client(client: Union[PromptsClient, AgentsClient]) -> FileType:
+    """Get the file type based on the client type."""
+    if isinstance(client, PromptsClient):   
+        return "prompt"    
+    elif isinstance(client, AgentsClient):
+        return "agent"
+    else:
+        raise ValueError(f"Unsupported client type: {type(client)}")
+
+def overload_with_local_files(
+    client: Union[PromptsClient, AgentsClient], 
+    sync_client: SyncClient,
+    use_local_files: bool,
+) -> Union[PromptsClient, AgentsClient]:
+    """Overload call and log methods to handle local files when use_local_files is True.
+    
+    When use_local_files is True, the following prioritization strategy is used:
+    1. Direct Parameters: If {file_type} parameters are provided directly (as a PromptKernelRequestParams or AgentKernelRequestParams object),
+       these take precedence and the local file is ignored.
+    2. Version/Environment: If version_id or environment is specified, the remote version is used instead
+       of the local file.
+    3. Local File: If neither of the above are specified, attempts to use the local file at the given path.
+    
+    For example, with a prompt client:
+    - If prompt={model: "gpt-4", ...} is provided, uses those parameters directly
+    - If version_id="123" is provided, uses that remote version
+    - Otherwise, tries to load from the local file at the given path
+    
+    Args:
+        client: The client to overload (PromptsClient or AgentsClient)
+        sync_client: The sync client used for file operations
+        use_local_files: Whether to enable local file handling
+        
+    Returns:
+        The client with overloaded methods
+        
+    Raises:
+        HumanloopRuntimeError: If use_local_files is True and local file cannot be accessed
+    """
+    original_call = client._call if hasattr(client, '_call') else client.call
+    original_log = client._log if hasattr(client, '_log') else client.log
+    file_type = _get_file_type_from_client(client)
+
+    def _overload(self, function_name: str, **kwargs) -> PromptCallResponse:
+        if "id" in kwargs and "path" in kwargs: 
+            raise HumanloopRuntimeError(f"Can only specify one of `id` or `path` when {function_name}ing a {file_type}")
+        # Handle local files if enabled
+        if use_local_files and "path" in kwargs:
+            # Check if version_id or environment is specified
+            has_version_info = "version_id" in kwargs or "environment" in kwargs
+            normalized_path = sync_client._normalize_path(kwargs["path"])
+            
+            if has_version_info:
```

Comment:
better name: use_workspace = any(["version_id" in kwargs, "environment" in kwargs])